src/journal-remote/journal-gatewayd.c

```diff
@@ -885,6 +888,139 @@ static int request_handler_machine(
         return MHD_queue_response(connection, MHD_HTTP_OK, response);
 }
 
+static int output_boot(FILE *f, LogId boot, int boot_display_index) {
+        int r;
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *json = NULL;
+
+        r = sd_json_build(
+                        &json,
+                        SD_JSON_BUILD_OBJECT(
+                                        SD_JSON_BUILD_PAIR("index", SD_JSON_BUILD_INTEGER(boot_display_index)),
+                                        SD_JSON_BUILD_PAIR_ID128("boot_id", boot.id),
+                                        SD_JSON_BUILD_PAIR("first_entry", SD_JSON_BUILD_UNSIGNED(boot.first_usec)),
+                                        SD_JSON_BUILD_PAIR("last_entry", SD_JSON_BUILD_UNSIGNED(boot.last_usec))));
+        if (r < 0)
+                return r;
+
+        r = sd_json_variant_dump(json, SD_JSON_FORMAT_SEQ, f, NULL);
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+static ssize_t request_reader_boots(
+                void *cls,
+                uint64_t pos,
+                char *buf,
+                size_t max) {
```

Comment:
Maybe better to rename `max` -> `buf_size` or so? Otherwise, it hides function `max()` if exists.

Reply:
OK. Then, feel free to keep it.